fix: set default header on isomorphic-git to fix BitBucket and AWS CodeCommit git sync

nHuebner opened the pull request:
### Context

When trying to do git operations with at least on premise BitBucket servers, the request could result in a 404.

### Possible Solution

Set deafult header for isomorphic-git httpClient.
Deafult behaviour for cli git command adds "Accept: */*" header,
to mimic this behaviour an defaultGitAcceptHeader will be set and merged with optional provided config headers.

Closes #2220 

Also,
Closes #4660
Closes INS-1811


Changelog(Fixes): Fixed an issue that prevented users from using Git Sync when trying to clone projects hosted in BitBucket Server and AWS CodeCommit

Commit:
fix: set default header f

Set deafult header for isomorphic-git httpClient.
Deafult behaviour for cli git command adds "Accept: */*" header,
to mimic this behaviour an defaultGitAcceptHeader will be set and merged with optional provided config headers.

#2220

filfreire reviewed (commented):
Hi @nHuebner, thank you a lot for contributing this!

I was trying to test out this locally using [Bitbucket OnPrem server (running in docker)](https://hub.docker.com/r/atlassian/bitbucket-server/#!), and I was able to 
- Create branches, push commits, ...

But, there might still be some problem. After initially pushing to another branch, and then merging it with the main branch I faced 2 problems:
- I couldn't pull latest changes once changing back to main branch
![Screenshot 2022-06-22 at 13 59 08](https://user-images.githubusercontent.com/11976836/175035893-c5902b66-e005-4773-95a0-e60f88da88b3.png)

- I couldn't clone the repo again (after deleting it, and trying to clone back)
![Screenshot 2022-06-22 at 14 00 18](https://user-images.githubusercontent.com/11976836/175035891-0cd02350-1f3d-4905-8b1f-dbef7fb6cb5d.png)

I'll need to test this out a bit more deeply. Did you try it out yourself and face any issues?

Commit:
fix: change default header behaviour

faulty behaviour of merge with config headers, so currently no merge with config.headers

filfreire commented:
@nHuebner heads up, I rebased the branch on top of latest develop.

Going to give it another try today and test to see if I can still reproduce the other problems. If not I think you should be able to merge it 👍

filfreire committed:
Fix bug where we always only send the Accept header, and miss other headers

filfreire reviewed (approved):
heads up @nHuebner I managed to re-test again this change locally, after rebasing with latest `develop`. I found a small bug which was basically the initial request no longer returned 404, but subsequent requests didn't carry the `config.headers` that had authorization related headers, making all git clone attempts fail with 401 error. After fixing this, most basic operations with Git against a Bitbucket server (in my case a local trial instance running in Docker) seem to work ok 👍 

![Screenshot 2022-08-25 at 14 27 19](https://user-images.githubusercontent.com/11976836/186678045-4e212b6a-a158-4549-b327-93d1ad7ab401.jpg)

gatzjames reviewed (approved):
Tested and works nice!
We might need to expose a UI to edit the headers in the future but I don't see a way this can break other behaviour!
Thanks a lot for the contribution!
